attiny.go

```diff
@@ -109,13 +110,12 @@ func (a *attiny) PingWatchdog() error {
 	return a.write(watchdogReg, nil)
 }
 
-func (a *attiny) checkIsOnBattery() error {
+func (a *attiny) checkIsOnBattery() (bool, error) {
 	batVal, err := a.readBatteryValue()
 	if err != nil {
-		return err
+		return false, err
 	}
-	a.onBattery = batVal > a.voltages.NoBattery
-	return nil
+	return batVal > a.voltages.NoBattery, nil
```

Comment:
Is it worth caching this result? The "am I on battery status" isn't going to change once it's powered on.

Keep returning the value as it is now, but use a cached value if it's been called before.